src/main/java/picard/sam/SamAlignmentMerger.java

```diff
@@ -158,6 +141,70 @@ public void mergeAlignment(final File referenceFasta) {
         }
     }
 
+    @Override
+    protected SAMSequenceDictionary getDictionaryForMergedBam() {
+        SAMSequenceDictionary finalDict = null;
+        if (alignedSamFile != null && !alignedSamFile.isEmpty()) {
+            for (final File alignedFile : alignedSamFile) {
+                final SAMSequenceDictionary dict = SAMSequenceDictionaryExtractor.extractDictionary(alignedFile);
+                if (finalDict == null) {
+                    finalDict = dict;
+                } else {
+                    if (!compareSequenceDictionaries(finalDict, dict)) {
+                        log.error("Dictionaries of input files differ. Unable to continue.");
+                        throw new UnsupportedOperationException("Cannot use input files with different dictionaries.");
+                    }
+                }
+            }
+        } else {
+            final SeparateEndAlignmentIterator mergingIterator = new SeparateEndAlignmentIterator(this.read1AlignedSamFile, this.read2AlignedSamFile, referenceFasta);
+            finalDict = mergingIterator.getHeader().getSequenceDictionary();
+        }
+
+        SAMSequenceDictionary referenceDict = SAMSequenceDictionaryExtractor.extractDictionary(referenceFasta);
+        if (referenceDict == null) {
+            throw new PicardException("No sequence dictionary found for " + referenceFasta.getAbsolutePath() +
+                    ".  Use CreateSequenceDictionary.jar to create a sequence dictionary.");
+        }
+        return mergeDictionaries(finalDict, SAMSequenceDictionaryExtractor.extractDictionary(referenceFasta));
+    }
+
+    private SAMSequenceDictionary mergeDictionaries(SAMSequenceDictionary dict1, SAMSequenceDictionary dict2) {
+
+        final SAMSequenceDictionary finalDict = new SAMSequenceDictionary(dict1.getSequences());
+        for (final SAMSequenceRecord sequence : finalDict.getSequences()) {
```

Comment:
So you make a copy of the dictionary, but then go through and modify the sequences in the dictionary, effectively modifying the source dictionary entries.  Why not just construct new records in the loop since you're setting all the attributes anyway?